src/main/java/frc/robot/subsystems/SwerveModule.java

```diff
@@ -0,0 +1,251 @@
+        boolean transInverse,
+        PIDController rotPID,
+        PIDController transPID
+    ) {
+        // Setting Parameters
+        _rotID = rotID;
+        _transID = transID;
+        _rotEncoderID = rotEncoderID;
+        _transEncoderID = transEncoderID;
+        _rotEncoderOffset = rotEncoderOffset;
+        _rotInverse = rotInverse;
+        _transInverse = transInverse;
+
+        // ----Setting Hardware
+        // Motor Controllers
+        _rotMotor = new CANSparkMax(_rotID, MotorType.kBrushless);
+        _transMotor = new CANSparkMax(_transID, MotorType.kBrushless);
+
+        // Encoders
+        _rotEncoder = new AnalogEncoder(_rotEncoderID);
+        _transEncoder = _transMotor.getEncoder();
+
+        
+        _rotEncoder.setPositionOffset(_rotEncoderOffset);
+
+
+        // ----Setting PID
+        _rotPID = rotPID;
+        _transPID = transPID;
+
+        // ----Setting PID Parameters
+        _rotPID.enableContinuousInput(-Math.PI,Math.PI);
+
+
+
+        // ----Setting Inversion
+        _rotMotor.setInverted(_rotInverse);
+        _transMotor.setInverted(_transInverse);
+    }
+
+    // ------------------- State Settings
+
+    /**
+     * 
+     * @return a swerve module state object describing the current speed of the translation and rotation motors
+     * @see SwerveModuleState
+     */
+    public SwerveModuleState getState() {
+        return new SwerveModuleState(getTransVelocity(), new Rotation2d(getRotPosition()));  
+    }
+
+        /**
+     * Sets the motor speeds passed into constructor
+     * @param desiredState takes in SwerveModule state
+     * @see SwerveModuleState
+     */
+    public void setDesiredState(SwerveModuleState desiredState, DriveMode dMode){
+        
+        // Stops returning to original rotation
+        if (Math.abs(desiredState.speedMetersPerSecond) < 0.001) {
+            stop();
+            return;
+        }
+
+        //No turning motors over 90 degrees
+        desiredState = SwerveModuleState.optimize(desiredState, getState().angle);
+
+        //PID Controller for both translation and rotation
+        switch(dMode){
+            case AUTO:
+                _transMotor.set(desiredState.speedMetersPerSecond/SwerveConstants.kPhysicalMaxSpeedMPS);
+                break;
+            case TELEOP:
+                _transMotor.set(desiredState.speedMetersPerSecond/SwerveConstants.kPhysicalMaxSpeedMPS);
+                break;
+        }
+
+        
+        _rotMotor.set(
+            _rotPID.calculate((-getRotPosition()) * 2 * Math.PI,
+                desiredState.angle.getRotations())
+        );
+  
+    }
+
+    /**
+     * 
+     * @return the total distance traveled by the module (Meters) and Rotation value (Rad) in the form of a SwerveModulePostion object
+     * @see SwerveModulePosition
+     */
+    public SwerveModulePosition getModulePos(){
+            return new SwerveModulePosition(getTransPosition(), new Rotation2d(getRotPosition()));
```

Comment:
The Rotation2d constructor takes in radians while the getRotPosition() method here returns rotations. Consider using the Rotation2d.fromRotations() method in order to quickly get the right Rotation2d object.

Also, the absolute encoder spins opposite to the motor, so like you did at the PIDController add a negative. Or, remove the negative at line 122 and make the getRotPosition method itself account for it.